lua/kubectl/views/pods/init.lua

```diff
@@ -26,40 +27,68 @@ function M.Draw(cancellationToken)
   root_definition.setPortForwards(M.builder.extmarks, M.builder.prettyData, M.pfs)
 end
 
-function M.TailLogs()
+function M.TailLogs(pod, ns, container)
+  pod = pod or M.selection.pod
+  ns = ns or M.selection.ns
+  container = container or M.selection.container
+  local ntfy = " tailing: " .. pod .. " container: " .. container
+  local args = { "logs", "--follow", "--since=1s", pod, "-n", ns, "-c", container }
   local buf = vim.api.nvim_get_current_buf()
-  vim.api.nvim_win_set_cursor(0, { vim.api.nvim_buf_line_count(buf), 0 })
+  local logs_win = vim.api.nvim_get_current_win()
+  vim.api.nvim_win_set_cursor(logs_win, { vim.api.nvim_buf_line_count(buf), 0 })
 
   local function handle_output(data)
     vim.schedule(function()
       if vim.api.nvim_buf_is_valid(buf) then
         local line_count = vim.api.nvim_buf_line_count(buf)
-
-        vim.api.nvim_buf_set_lines(buf, line_count, line_count, false, vim.split(data, "\n"))
+        vim.api.nvim_buf_set_lines(buf, line_count, line_count, false, vim.split(data, "\n", { trimempty = true }))
         vim.api.nvim_set_option_value("modified", false, { buf = buf })
-        vim.api.nvim_win_set_cursor(0, { line_count + 1, 0 })
+        if logs_win == vim.api.nvim_get_current_win() then
+          vim.api.nvim_win_set_cursor(0, { line_count + 1, 0 })
+        end
       end
     end)
   end
 
-  local args = { "logs", "--follow", "--since=1s", M.selection.pod, "-n", M.selection.ns }
-  local handle = commands.shell_command_async("kubectl", args, nil, handle_output)
+  local function stop_tailing(handle)
+    handle:kill(2)
+    vim.notify("Stopped" .. ntfy, vim.log.levels.INFO)
+  end
+
+  local group = vim.api.nvim_create_augroup("__kubectl_tailing", { clear = false })
+  if M.tail_handle and not M.tail_handle:is_closing() then
+    vim.api.nvim_clear_autocmds({ group = group })
+    stop_tailing(M.tail_handle)
+  else
+    M.tail_handle = commands.shell_command_async("kubectl", args, nil, handle_output)
 
-  vim.notify("Start tailing: " .. M.selection.pod, vim.log.levels.INFO)
-  vim.api.nvim_create_autocmd("BufWinLeave", {
-    buffer = buf,
-    callback = function()
-      handle:kill(2)
-      vim.notify("Stopped tailing: " .. M.selection.pod, vim.log.levels.INFO)
-    end,
-  })
+    vim.notify("Started " .. ntfy, vim.log.levels.INFO)
+    vim.api.nvim_create_autocmd("BufWinLeave", {
+      group = group,
+      buffer = buf,
+      callback = function()
+        stop_tailing(M.tail_handle)
+      end,
+    })
+  end
 end
 
-function M.selectPod(pod_name, namespace)
-  M.selection = { pod = pod_name, ns = namespace }
+function M.selectPod(pod, ns)
+  local container
+  local data = M.builder.data.items
```

Comment:
I'm a bit sceptical about this approach, what are we trying to do here?

Reply:
if you follow logs for a pod, kubectl will select the first container automagically
https://kubernetes.io/docs/reference/kubectl/generated/kubectl_logs/
according to this we can add the flag: `--all-containers=true`
so it needs to be refactored this way.
Unfortunately I don't have any time today and tomorrow i'm going on a vacation for a week.
Please feel free to step in on this branch and do whatever you think.
I think toggling follow is an important feature.
before this, I would need to leave the log view in order to stop following and sometimes I just follow logs and just want to stop and investigate the output.

every feature I add is out of real life usage of the plugin :)

Reply:
I agree and appreciate your effort a lot ❤️, wasn't questioning your motives nor the idea!
Was trying to balance the feature against performance and sustainability and a gut feeling that we could do something better here but wanted to know if I've missed something that would make this a valid approach.
We can merge and have that as a single fix up issue ✌️

Reply:
And also! Have a nice vacation! Well deserved IMHO!!